src/test/java/net/snowflake/client/core/SessionUtilLatestIT.java

```diff
@@ -390,4 +390,79 @@ public void testOktaAuthGetFail() throws Throwable {
       assertEquals(SqlState.IO_ERROR, e.getSQLState());
     }
   }
+
+  private SFLoginInput createOktaLoginInput() {
+    SFLoginInput input = new SFLoginInput();
+    input.setServerUrl("https://testauth.okta.com");
+    input.setUserName("MOCK_USERNAME");
+    input.setPassword("MOCK_PASSWORD");
+    input.setAccountName("MOCK_ACCOUNT_NAME");
+    input.setAppId("MOCK_APP_ID");
+    input.setOCSPMode(OCSPMode.FAIL_OPEN);
+    input.setHttpClientSettingsKey(new HttpClientSettingsKey(OCSPMode.FAIL_OPEN));
+    input.setLoginTimeout(1000);
+    input.setSessionParameters(new HashMap<>());
+    input.setAuthenticator("https://testauth.okta.com");
+    return input;
+  }
+
+  // Testing retry with Okta calls the service to get a new unique token. This is valid after
+  // version 3.15.0.
+  @Test
+  public void testOktaAuthRetry() throws Throwable {
+    SFLoginInput loginInput = createOktaLoginInput();
+    Map<SFSessionProperty, Object> connectionPropertiesMap = initConnectionPropertiesMap();
+    SnowflakeSQLException ex =
+        new SnowflakeSQLException(ErrorCode.AUTHENTICATOR_REQUEST_TIMEOUT, 0, true, 0);
```

Comment:
this exception is thrown when timeouts are reached - what about testing 429 error code? I think that there should be a test with NETWORK_ERROR and code 429 thrown during execution

Reply:
Everything around checking retry related to the 429 error code happens within [RestRequest.execute()](https://github.com/snowflakedb/snowflake-jdbc/blob/557659ed6ebdd8a7acfd6349a5f6d632bb833502/src/main/java/net/snowflake/client/jdbc/RestRequest.java#L110). When it detects a 429 error code it eventually throws the AUTHENTICATOR_REQUEST_TIMEOUT [here](https://github.com/snowflakedb/snowflake-jdbc/blob/557659ed6ebdd8a7acfd6349a5f6d632bb833502/src/main/java/net/snowflake/client/jdbc/RestRequest.java#L402) which is what we're testing for in this case. I've tried to see if I could set up the test to show the whole flow from a 429 error being thrown then eventually leading to getting a new Okta token, but the mocking set up gets very quickly complicated because of the level of mocks you need to set up. So what I've done instead is I've added a new test in RestRequestTest to show that it throws the AUTHENTICATOR_REQUEST_TIMEOUT when it encounters a 429 error. 

Does that seem like enough test support?